chapter03/Task04.py

```diff
@@ -7,17 +8,26 @@
       '"=" - Если я ответил верно')
 
 number = 50
+a = 0
+b = 101
+con = 0
 
 ans = str()
 while True:
-    ans = str(input('Это число' + str(number)))     #Внести правки по поводу диапозонов, проверки кода, дописать прогу
+    ans = input('Это число ' + str(number) + '?\n')
+    con += 1
     if ans == '=':
         break
     elif ans == '-':
-        number //= 2
+        b = number
+        number = b - (b - a) // 2
```

Comment:
Seems like you have different formulas for "number" that doesn't really depend on if it was "+" or "-" before, but more on just what are "a" and "b" values. Consider moving "number" computation into a single place.